### PR TITLE
feat(tags): Editing Tags Analyltics [IN-927]

### PR DESCRIPTION
**Pull request by timc-mozilla**
## Summary
* Analytics for edit tags

## References 
* [IN-927](https://getpocket.atlassian.net/jira/software/projects/IN/boards/80/backlog?assignee=62f66d8bb5b801a9aff1f896&selectedIssue=IN-927)

## Implementation Details
* added analytics for edit tags on Saves and Archive

## Test Steps
* analytics test docker
* test Batch 3 part 2 [here](https://getpocket.atlassian.net/wiki/spaces/PFC/pages/2559836224/iOS+Next+Analytics#%231---engagement-events-in-Filtering-Tags-flow)

## PR Checklist:
- [x] Added Unit / UI tests
- [x] Self Review (review, clean up, documentation, run tests)
- [x] Basic Self QA

### CODE REVIEW

**Comment by dskuza** [comment]
@timc-mozilla @cyndichin I'm a little confused by what's in the document, so I'm looking for clarification:

Under `Deleting a Tag from overflow`:

```
[
  {
    "hierarchy": 0,
    "identifier": "tagsOverflow",
    "type": "button"
  }
]
[
  {
    "hierarchy": 0,
    "identifier": "tagsDelete",
    "type": "button"
  }
]
```

Does this mean that we are supposed to send _two_ engagement events, and that is the hierarchy for each? I'm missing why there are two arrays with one context.

**Comment by cyndichin** [comment]
> @timc-mozilla @cyndichin I'm a little confused by what's in the document, so I'm looking for clarification:
> 
> Under `Deleting a Tag from overflow`:
> 
> ```
> [
>   {
>     "hierarchy": 0,
>     "identifier": "tagsOverflow",
>     "type": "button"
>   }
> ]
> [
>   {
>     "hierarchy": 0,
>     "identifier": "tagsDelete",
>     "type": "button"
>   }
> ]
> ```
> 
> Does this mean that we are supposed to send _two_ engagement events, and that is the hierarchy for each? I'm missing why there are two arrays with one context.



> @timc-mozilla @cyndichin I'm a little confused by what's in the document, so I'm looking for clarification:
> 
> Under `Deleting a Tag from overflow`:
> 
> ```
> [
>   {
>     "hierarchy": 0,
>     "identifier": "tagsOverflow",
>     "type": "button"
>   }
> ]
> [
>   {
>     "hierarchy": 0,
>     "identifier": "tagsDelete",
>     "type": "button"
>   }
> ]
> ```
> 
> Does this mean that we are supposed to send _two_ engagement events, and that is the hierarchy for each? I'm missing why there are two arrays with one context.

Yes, every array is for one engagement. One is for clicking the overflow / Edit button and the other event would be for the delete button. I copied what was in the docs, but I can format it in a better way if that helps.

File: PocketKit/Sources/PocketKit/MyList/ArchivedItemsList/ArchivedItemsListViewModel.swift [hunk 1]
```diff
@@ -390,8 +390,10 @@ extension ArchivedItemsListViewModel {
                 case .all:
                     return nil
                 case .tagged:
+                    filterTagOverflowAnalytics()
```
Comment (dskuza) [inline PocketKit/Sources/PocketKit/MyList/ArchivedItemsList/ArchivedItemsListViewModel.swift:393]: 🤔 Something about this seems incorrect. It's not exactly an overflow button (compared to the overflow button in an item itself). @cyndichin double checking here - what's correct for when "Tagged" is selected in the carousel?

Looking at the doc, though, it does appear that there's just a basic UI hierarchy of overflow button -> list screen (either my_list or archive). So… is this just an overloaded word?

**Review by dskuza — APPROVED** [review]
I think based on the docs, this is correct, but I do have a question about a quirk in the requirements. Other than that, @cyndichin you've been point for a lot of the tags analytics so I'm going to lean on you here for a second opinion.

File: PocketKit/Sources/PocketKit/MyList/ArchivedItemsList/ArchivedItemsListViewModel.swift (hunk 1 above)
Comment (timc-mozilla) [reply PocketKit/Sources/PocketKit/MyList/ArchivedItemsList/ArchivedItemsListViewModel.swift:393]: Yeah, you're right. i think I misunderstood the context of the analytics. ill fix that up later today.

**Comment by timc-mozilla** [comment]
Merging PRs with 926

**Pull request closed without merging**